extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -189,12 +190,15 @@ export class AmpAnalytics extends AMP.BaseElement {
             trigger['selector'] = this.expandTemplate_(trigger['selector'],
                 trigger, /* arg*/ undefined, /* arg */ undefined,
                 /* arg*/ false);
-            this.instrumentation_.addListener(
-                trigger, this.handleEvent_.bind(this, trigger), this.element);
-
+            this.instrumentationPromise_.then(instrumentation => {
```

Comment:
Yes. @avimehta's suggestion is better. There won't be any real layout delays. But what we want is to resolve instrumentation _after_ we start fetch.